kie-server-parent/kie-server-services/kie-server-services-dmn/src/main/java/org/kie/server/services/dmn/ModelEvaluatorServiceBase.java

```diff
@@ -204,8 +217,128 @@ public static DMNItemDefinitionInfo itemDefinitionToInfo(ItemDefinition itemDef)
         }
     }
 
+    public Response evaluateModel(String containerId, String modelId, String contextPayload, boolean asDmnResult, String decisionServiceId) {
+        try {
+            KieContainerInstanceImpl kContainer = context.getContainer(containerId, ContainerLocatorProvider.get().getLocator());
+            DMNRuntime dmnRuntime = KieRuntimeFactory.of(kContainer.getKieContainer().getKieBase()).get(DMNRuntime.class);
+
+            List<DMNModel> modelsWithID = dmnRuntime.getModels().stream().filter(m -> m.getName().equals(modelId)).collect(Collectors.toList());
+            if (modelsWithID.isEmpty()) {
+                return Response.serverError().entity("No model identifies with modelId: " + modelId).build();
+            } else if (modelsWithID.size() > 1) {
+                return Response.serverError().entity("More than one existing DMN model having modelId: " + modelId).build();
+            }
+            DMNModel dmnModel = modelsWithID.get(0);
+            DecisionServiceNode determinedDS = null;
+            if (decisionServiceId != null) {
+                Optional<DecisionServiceNode> dsOpt = dmnModel.getDecisionServices().stream().filter(ds -> ds.getName().equals(decisionServiceId)).findFirst();
+                if (!dsOpt.isPresent()) {
+                    return Response.serverError().entity("No decisionService found: " + decisionServiceId).build();
```

Comment:
Is it meaningful to return HTTP 500 here? I would say that we should return 404 as this is not server error but rather request issue.

Reply:
I don't like it very much either, but for consistency with `kie-server` generic endpoints I've made it be server error.
If you would approve of a 404, so diverging sligthly in semantic considering these are "new kind of" endpoints, I'm happy to change it.

Please let me know.

Reply:
I would prefer the 404, the divergence is ok for me as those are new endpoints.